comms/core/src/peer_manager/manager.rs

```diff
@@ -146,15 +146,16 @@ impl PeerManager {
         self.peer_storage_sql.all(features)
     }
 
-    /// Get available dial candidates that are communication nodes, not banned, not deleted,
+    /// Get available dial candidates that are communication nodes, not banned, not deleted, reachable
     /// and not in the excluded node IDs list
     pub async fn get_available_dial_candidates(
         &self,
         exclude_node_ids: &[NodeId],
         limit: Option<usize>,
+        transport_protocols: &[TransportProtocol],
```

Comment:
```suggestion
        transport_protocols: &[TransportProtocol],
```
Cant we make this a self varialbe, then we dont need to ask for it each time from the callee?

Reply:
I do not think `transport_protocols` in this context is something that `PeerManager` should own - it is determined by the caller.